lib/writers/s3.go

```diff
@@ -66,15 +66,17 @@ func (b *S3Writer) WriteDirWithObjects(deleteExistingContentsInDir bool, dir str
 		var err error
 		objectsToDeleteMap, err = b.getObjectsInDir(ctx, dir, logger)
 		if err != nil {
+			logger.Error(err, "Error getting objects to delete")
 			return err
 		}
 	}
 
 	exists, errBucketExists := b.RepoClient.BucketExists(ctx, b.BucketName)
 	if errBucketExists != nil {
 		logger.Error(errBucketExists, "Could not verify bucket existence with provider")
+		return errBucketExists
 	} else if !exists {
-		logger.Info("Bucket provided does not exist (or the provided keys don't have permissions)")
+		return fmt.Errorf("Bucket provided does not exist (or the provided keys don't have permissions)")
 	}
 
```

Comment:
previously, the bucket not existing (or our inability to verify) wouldn't raise an error. We are now raising it and returning. Is this desirable? 🤔

Reply:
Yes I believe so. If we don't return an error here we will continue on to try and write the object when the bucket doesn't exist. By returning the error here we have the ability to requeue and wait for the bucket to be accessible.